### ISSUE TITLE
Replace 'whitelist' and 'blacklist' with appropriate 'allow' and 'deny' list names.

### ISSUE THREAD

**Issue by howbazaar** [cloud-custodian/cloud-custodian#6105]
There are 52 references to 'blacklist' and 162 references to 'whitelist' in the codebase.

An example is in `sns.py` where `whitelist` is used as a filter option on `cross-account`.

Is there a way to deprecate schema attributes? To have things produce informational info on validation, but still work?

**Comment by kapilt** [comment]
yeah i'd like to switch the documented version to allowed/denied but we'll need to keep compatibility for extant policies. see other issues re deprecation mechanism (still needs a flushed out spec) and impl.
